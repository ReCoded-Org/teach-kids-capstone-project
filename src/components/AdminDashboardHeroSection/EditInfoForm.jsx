import React from "react";
import "flowbite";

export default function EditInfoForm({showModal, setShowModal, handleChange, dataSender}) {

    return (
        <>
            <div className='my-6  flex justify-end gap-6 md:justify-end '>
                <button
                    onClick={() => setShowModal(true)}
                    type='button'
                    href=''
                    className='mw-40 w-36 rounded border border-[#457B9D] bg-[#457B9D]  p-1.5 font-semibold text-white  duration-300 ease-linear hover:rounded  hover:border-blue-btn hover:bg-transparent hover:text-[#457B9D] hover:shadow md:w-44'
                >
                    Edit
                </button>
            </div>

            {showModal ? (
                <>
                    <div className=' fixed  inset-0 z-50 flex items-center justify-center overflow-y-auto overflow-x-hidden outline-none focus:outline-none'>
                        <div className='relative my-6 mx-auto w-auto max-w-3xl'>
                            {/*content*/}
                            <div className='relative flex w-full flex-col rounded-lg border-0 bg-white shadow-lg outline-none focus:outline-none'>
                                {/*header*/}
                                <div className='flex items-start justify-between rounded-t border-b border-solid border-slate-200 p-5'>
                                    <h3 className='text-3xl font-semibold'>
                                        Edit Pofile
                                    </h3>
                                    <button
                                        className='float-right ml-auto border-0 bg-transparent p-1 text-3xl font-semibold leading-none text-black opacity-50 outline-none focus:outline-none'
                                        onClick={() => setShowModal(false)}
                                    >
                                        <span className='block h-6 w-6 bg-transparent text-2xl text-black outline-none focus:outline-none'>
                                            X
                                        </span>
                                    </button>
                                </div>
                                {/*body*/}
                                <div className='relative h-[27rem] w-[19rem] flex-auto overflow-y-auto p-6 sm:h-[33rem] sm:w-[35rem]'>
                                    {/* START OF FORM */}

                                    <form className='mb-4 rounded bg-white px-8 pt-6 pb-8 shadow-md'>
                                        <div className='mb-4'>
                                            <label className='text-gray-700 mb-2 block text-sm font-bold'>
                                                Select the Event Photo
                                            </label>
                                            <input
                                                className='text-gray-700 focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight shadow focus:outline-none'
                                                type='file'
                                                placeholder=''
                                            />
                                        </div>
                                        <div>
                                            <label className='text-gray-700 mb-2 block text-sm font-bold'>
                                                Website Address :
                                            </label>
                                            <input
                                                onChange={handleChange}
                                                name='website'
                                                className='text-gray-700 focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight shadow focus:outline-none'
                                                type='text'
                                                placeholder='www.reach.com'
                                            />
                                        </div>
                                        <div className='mb-4'>
                                            <label className='text-gray-700 mt-3 mb-2 block text-sm font-bold'>
                                                Name
                                            </label>
                                            <input
                                                onChange={handleChange}
                                                name='name'
                                                className='text-gray-700 focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight shadow focus:outline-none'
                                                type='text'
                                                placeholder='John Doe'
                                            />
                                        </div>

                                        <div className='mb-4'>
                                            <label className='text-gray-700 mb-2 block text-sm font-bold'>
                                                Location
                                            </label>
                                            <input
                                                onChange={handleChange}
                                                name='location'
                                                className='text-gray-700 focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight shadow focus:outline-none'
                                                type='text'
                                                placeholder='Istanbul'
                                            />
                                        </div>

                                        <label className='text-gray-700 mb-2 block text-sm font-bold'>
                                            Date:
                                        </label>
                                        <div className='relative mb-4'>
                                            <div className='pointer-events-none absolute inset-y-0 left-0 flex items-center pl-3'>
                                                <svg
                                                    aria-hidden='true'
                                                    className='text-gray-500 dark:text-gray-400 h-5 w-5'
                                                    fill='currentColor'
                                                    viewBox='0 0 20 20'
                                                    xmlns='http://www.w3.org/2000/svg'
                                                >
                                                    <path
                                                        fillRule='evenodd'
                                                        d='M6 2a1 1 0 00-1 1v1H4a2 2 0 00-2 2v10a2 2 0 002 2h12a2 2 0 002-2V6a2 2 0 00-2-2h-1V3a1 1 0 10-2 0v1H7V3a1 1 0 00-1-1zm0 5a1 1 0 000 2h8a1 1 0 100-2H6z'
                                                        clipRule='evenodd'
                                                    />
                                                </svg>
                                            </div>
                                            <input
                                                onChange={handleChange}
                                                name='date'
                                                datepicker
                                                datepicker-autohide
                                                type='text'
                                                className='bg-gray-50 border-gray-300 text-gray-900 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 block w-full rounded-lg border p-2.5 pl-10  focus:border-blue-500 focus:ring-blue-500 dark:text-white dark:focus:border-blue-500 dark:focus:ring-blue-500 sm:text-sm'
                                                placeholder='DD-MM-YYYY'
                                            />
                                        </div>

                                        <div className='mb-4'>
                                            <label className='text-gray-700 mb-2 block text-sm font-bold'>
                                                Email
                                            </label>
                                            <input
                                                onChange={handleChange}
                                                name='email'
                                                className='text-gray-700 focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight shadow focus:outline-none'
                                                type='text'
                                                placeholder='john.doe@gamil.com'
                                            />
                                        </div>

                                        <div className='mb-4'>
                                            <label className='text-gray-700 mb-2 block text-sm font-bold'>
                                                Phone Number
                                            </label>
                                            <input
                                                onChange={handleChange}
                                                name='phone'
                                                className='text-gray-700 focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight shadow focus:outline-none'
                                                type='tel'
                                                placeholder='+90 (545) 121 3333'
                                            />
                                        </div>

                                        <div className=''>
                                            <label className='text-gray-700 mb-2 block text-sm font-bold'>
                                                Your Message
                                            </label>
                                            <div className='form-group'>
                                                <input
                                                    name='message'
                                                    onChange={
                                                        handleChange
                                                    }
                                                    className='
                            form-control text-gray-700 border-gray-300 focus:text-gray-700 m-0 block w-full rounded border border-solid bg-white bg-clip-padding px-3 py-1.5 text-base font-normal transition
                            ease-in-out focus:border-blue-600 focus:bg-white focus:outline-none
                        '
                                                    id='exampleFormControlTextarea13'
                                                    rows='3'
                                                    placeholder='Message'
                                                ></input>
                                            </div>
                                        </div>
                                    </form>
                                    {/* END OF FORM */}
                                </div>
                                {/*footer of the form*/}
                                <div className='flex items-center justify-end rounded-b border-t border-solid border-slate-200 py-2 px-6'>
                                    <button
                                        className='background-transparent mr-1 mb-1 px-6 py-2 text-sm font-bold uppercase text-black outline-none transition-all duration-150 ease-linear focus:outline-none'
                                        type='button'
                                        onClick={() => setShowModal(false)}
                                    >
                                        Close
                                    </button>
                                    <button
                                        className='mr-1 mb-1 rounded bg-[#457B9D] px-6 py-3 text-sm font-bold uppercase text-white shadow outline-none transition-all duration-150 ease-linear hover:shadow-lg focus:outline-none active:border-2 active:border-[#457B9D] active:bg-white active:text-[#457B9D]'
                                        type='submit'
                                        onClick={() => {
                                            setShowModal(false);
                                            dataSender();
                                        }}
                                    >
                                        Save Changes
                                    </button>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div className='fixed inset-0 z-40 bg-black opacity-25'></div>
                </>
            ) : null}
        </>
    );
}
