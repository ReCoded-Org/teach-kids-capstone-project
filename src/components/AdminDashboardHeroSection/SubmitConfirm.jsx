import React from 'react';

export default function SubmitConfirm () {
  return (
    <div>
    {/* Main modal */}
    <div id="defaultModal" tabIndex={-1} className="overflow-y-auto overflow-x-hidden fixed top-0 right-0 left-0 z-50 w-full md:inset-0 h-modal md:h-full justify-center items-center flex" aria-modal="true" role="dialog">
      <div className="relative p-4 w-100 max-w-2xl h-full md:h-auto">
        {/* Modal content */}
        <div className="relative bg-customGreen rounded-lg shadow dark:bg-gray-700">
          {/* Modal header */}
          <div className="flex justify-between items-start p-4 ">
            <h3 className="text-xl font-semibold text-gray-900 dark:text-white">
              Saved
            </h3> 
          </div>
          {/* Modal body */}
                
        </div>
      </div>
    </div>
  </div>
    )



}