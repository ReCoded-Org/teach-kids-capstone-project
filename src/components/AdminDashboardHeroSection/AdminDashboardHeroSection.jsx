import React, { useState } from "react";
import "../../App.css";
import NGOImage from "../../assets/theBluesProfile.png";
import NGOEvent from "../../assets/theBluesEvents.jpg";
import "../../App.css";
import EditInfoForm from "./EditInfoForm";
import { useQuery } from "@tanstack/react-query";

function AdminDashboardHeroSection({ userId }) {
    const [info, setInfo] = useState([]); //changed {} to []
    const { isLoading, error, data } = useQuery(["repoData"], () =>
        fetch(`https://reach-capstone.herokuapp.com/api/ngos/${userId}`).then(
            (res) => res.json().then((data) => setInfo(data.data))
        )
    );
    if (isLoading) return "Loading...";

    // if (error) return error.messag;
    return (
        <>
            <section
                className={
                    "-mb-12   flex h-[60rem] flex-col   justify-around  gap-6 bg-gray p-6 pb-12 md:px-20 lg:flex-row-reverse  lg:px-10 xl:px-40 "
                }
            >
                <main className='relative mt-56 md:mr-6 lg:mt-6 lg:w-10/12 xl:w-2/4  '>
                    <span>
                        <EditInfoForm userId={userId} />
                    </span>
                    <div className='my-6  flex justify-end gap-6 md:justify-end '></div>
                    <div className='row1 grid md:flex md:flex-row-reverse  '>
                        <img
                            src={NGOImage}
                            className=' h-40 w-40 justify-self-center rounded-full border-8 border-white '
                            alt=''
                        />
                        <ul className='mt-4 grid w-full gap-1.5 justify-self-auto text-left font-SourceSansPro text-sm font-semibold text-light-gray'>
                            <h1 className=' my-2 mb-4 justify-self-center font-quicksand  text-4xl font-semibold uppercase text-blue-dark md:justify-self-start'>
                                {info.name}
                            </h1>
                            <li>Location: {info.location}</li>
                            <li>Date: {info.createdAt.slice(0, 10)}</li>
                            <li>Email: {info.email}</li>
                            <li>Website: {info.website}</li>
                            <li>Phone: +{info.phone}</li>
                        </ul>
                    </div>
                    <div className='row2 my-4 grid text-justify'>
                        <p className='text-md my-2 font-semibold  text-blue-dark'>
                            {info.message}The goal of Save the Children is to
                            promote global education and the rights of children
                            around the world. To increase the quality of
                            instruction and help ensure lasting education, Save
                            the Children teaches effective teaching strategies
                            to instructors and trains them to engage students.
                            They coach parents and caregivers to help foster
                            learning early on, and offer ways for parents to
                            encourage schoolwork and continued learning outside
                            of the classroom. They also hope to introduce
                            children to artistic expression, encourage learning
                            during and after crisis and invest in the health of
                            children to ensure they donâ€™t fall behind. In 2012,
                            Save the Children reached 9 million children
                        </p>
                        <a
                            href={"https://" + info.website}
                            className='font-light-gray mx-4 mt-2 mb-4 justify-self-end font-semibold'
                        >
                            Read more on our website ...
                        </a>
                    </div>
                </main>
                <div className='image  grid place-items-center  md:p-4'>
                    <img
                        src={NGOEvent}
                        className=' w-3/4 items-center justify-self-center rounded-lg border-black object-contain shadow-lg  md:w-10/12  lg:justify-self-start	  '
                        alt=''
                    />
                </div>
            </section>
        </>
    );
}

export default AdminDashboardHeroSection;
