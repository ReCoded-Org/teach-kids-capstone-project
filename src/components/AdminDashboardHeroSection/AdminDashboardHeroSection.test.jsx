import React from "react";
import renderer from "react-test-renderer";
import AdminDashboardHeroSection from "./AdminDashboardHeroSection";
import { MemoryRouter } from "react-router-dom";

it("renders correctly", () => {
    const test = renderer
        .create(
            <MemoryRouter>
                <AdminDashboardHeroSection />
            </MemoryRouter>
        )
        .toJSON();
    expect(test).toMatchSnapshot();
});
