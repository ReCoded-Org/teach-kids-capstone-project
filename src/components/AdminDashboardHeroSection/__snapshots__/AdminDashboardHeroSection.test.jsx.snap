// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="flex  flex-col justify-around gap-6  bg-gray p-6 pb-12 md:px-20 lg:flex-row-reverse lg:gap-12 lg:px-10 xl:px-40 "
>
  <main
    className="md:mr-6 lg:w-2/3 xl:w-2/4 "
  >
    <div
      className="my-6  flex justify-end gap-6 md:justify-end "
    >
      <button
        className="mw-40 hover:bg-transparent w-36 border border-blue-btn  bg-blue-btn p-1.5 font-semibold  text-white duration-300 ease-linear  hover:rounded hover:border-blue-btn hover:text-blue-btn  hover:shadow md:w-44"
        href=""
        onClick={[Function]}
      >
        Edit
      </button>
    </div>
    <div
      className="row1 grid md:flex md:flex-row-reverse  "
    >
      <img
        alt=""
        className=" h-40 w-40 justify-self-center rounded-full border-8 border-white "
        src="NGOImage.png"
      />
      <ul
        className="mt-4 grid w-full gap-1.5 justify-self-auto text-left font-SourceSansPro text-sm font-semibold text-light-gray"
      >
        <h1
          className=" my-2 mb-4 justify-self-center  font-quicksand text-4xl font-semibold text-blue-dark md:justify-self-start"
        >
          Starnation
        </h1>
        <li>
          Location: Istanbul
        </li>
        <li>
          Date: 28-03-2023
        </li>
        <li>
          Email: helloworld@gmail.com
        </li>
        <li>
          Website: www.helloworld.com
        </li>
        <li>
          Phone: +90 535 898 54 45
        </li>
      </ul>
    </div>
    <div
      className="row2 my-4 grid text-justify"
    >
      <p
        className="text-md my-2 font-semibold  text-blue-dark"
      >
        It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like)......
      </p>
      <a
        className="font-light-gray mx-4 mt-2 mb-4 justify-self-end font-semibold"
        href="#"
      >
        Read more ...
      </a>
    </div>
  </main>
  <div
    className="image grid place-items-center  md:p-4"
  >
    <img
      alt=""
      className="md:w-4/4  items-center justify-self-center rounded-lg object-contain  lg:justify-self-start	  "
      src="NGOImage.png"
    />
  </div>
</section>
`;
